src/App.tsx

```
@@ -1,27 +1,56 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
+import * as goodsAPI from './api/goods';
 
 // import { getAll, get5First, getRed } from './api/goods';
 // or
 // import * as goodsAPI from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const loadAll = async () => {
+    try {
+      const data = await goodsAPI.getAll();
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+      setGoods(data);
+    } catch (error) {}
```

It's great that you've added `try...catch` blocks as requested! However, leaving the `catch` block empty means that if an error occurs, it will be silently ignored. This makes debugging very difficult.

To properly handle the error, you should at least log it to the console. This will help you identify problems during development. For example:

```javascript
} catch (error) {
  console.error('Failed to load data:', error);
}
```

This same feedback applies to the other `catch` blocks on lines 27 and 35.